src/Expr.h

```diff
@@ -937,14 +943,40 @@ class CallExpr : public Expr {
 
 protected:
 	friend class Expr;
-	CallExpr()	{ func = 0; args = 0; }
+	CallExpr() { func = 0; args = 0; }
 
 	void ExprDescribe(ODesc* d) const override;
 
 	Expr* func;
 	ListExpr* args;
 };
 
+/*
+	Class to handle the creation of anonymous functions with closures.
+
+	Facts:
+		- LambdaExpr creates a new BroFunc on every call to Eval.
+		- LambdaExpr must be given all the information to create a BroFunc on
+		  construction except for the closure.
+		- The closure for created BroFuncs is the frame that the LambdaExpr is
+		  evaluated in.
+*/
```

Comment:
Really minor nit - the comment formatting does not match with how we typically format comments.

Reply:
I tried to clean this up a little, although I am a little unclear on what proper comment formatting is for long comments like that is. Seemed like there is pretty minimal usage of multi-line comments, so I went ahead and replaced it with single line ones. Let me know if that's the wrong style as well.

Reply:
Generally Doxygen style is what we've been using for recent code.